### PR TITLE
Feature/2058/updated check mark

### PR DESCRIPTION
**Pull request by ByteMap**
Added Verified column to the entries tables

dockstore/dockstore#2058

Added a Verified column that is center justified for the check marks as 
well as adjusted the sizes of the column spacing for the entry tables in 
the homepage and in the search page. 

I have also made the star column right justified so that the elements are 
aligned with the stars and it would avoid any messiness when the entries'
number of digits next to the star icon begins to differentiate 

I was unable to get the sorting done due to the issue where I cannot
get a query sorted by verify in the back end.
Screen Shot 2019-08-27 at 3 37 24 PM
Screen Shot 2019-08-27 at 3 37 00 PM
Screen Shot 2019-08-27 at 3 37 06 PM
Screen Shot 2019-08-27 at 3 37 19 PM

### CODE REVIEW

File: src/app/containers/list/list.component.html [hunk 1]
```diff
@@ -26,38 +26,47 @@
     </mat-form-field>
   </div>
   <mat-table class="mat-elevation-z4" [dataSource]="dataSource" matSort matSortActive="stars" matSortDirection="desc" matSortDisableClear>
-    <ng-container matColumnDef="name">
-      <mat-header-cell *matHeaderCellDef mat-sort-header>Name</mat-header-cell>
-      <mat-cell *matCellDef="let tool">
+    <ng-container matColumnDef="check_mark">
```
[the remark this answers is not in the document]
Comment (denis-yuen) [reply src/app/containers/list/list.component.html]: It does look a bit spacious. 
I wonder if we should change the GitHub, quay.io, Docker Hub links to b/w icons , but that's a ticket for a different time.

**Comment by garyluu** [comment]
Ok, couldn't exactly put my finger on it.  But I think I finally figured out why.

It is deviating more from the Material guide: https://material.io/components/data-tables/#anatomy

1. Material suggests the padding between columns should be 32dp or more.  
In my opinion, it should just be 32px.  Though it's an implementation detail that should really matter, I prefer it to be 16px on either side of each column for symmetry.  To me, a column's size (without padding) is equivalent to the min size that fits both the content and header name.  If the viewport has too much horizontal space, give the extra space to name.  

2. Material suggests truncating with an ellipsis.
The break-word may seem nice at first, but some guy is going to make tools with a super long names and then get all his friends to star them to the top. The table would be really long.  Another reason is that the pagination buttons gets shifted up and down depending on the content which annoys some people.

Personal:

3. Even though Material doesn't specifically mention consistent padding, I prefer it (and maybe it's also implied).  I mentioned this before, but I find the 25px padding and 30px padding to be odd.  Specifically, the space between the "d" of "Verified" and the "A" of "Author" has about 70px padding on my screen. Whereas the space between the longest content in the Author column to the "F" of "Format" has 25px.  Making the padding from 70px to 32px should give more room to the name column that we desperately need.

File: src/app/containers/list/list.component.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (coverbeck) [reply src/app/containers/list/list.component.html]: > I wonder if we should change the GitHub, quay.io, Docker Hub links to b/w icons , but that's a ticket for a different time.

I like that idea a lot; icons will both save space and look better.

File: src/app/containers/list/list.component.html [hunk 2]
```diff
@@ -40,6 +37,14 @@
         }}</a>
       </mat-cell>
     </ng-container>
+    <ng-container matColumnDef="verified">
+      <mat-header-cell *matHeaderCellDef>Verified</mat-header-cell>
+      <mat-cell *matCellDef="let tool">
+        <a *ngIf="getVerified(tool)" href="{{ verifiedLink }}">
```
Comment (garyluu) [inline src/app/containers/list/list.component.html]: ugh, I need to fix this at some point

**Comment by coverbeck** [comment]
@ByteMap , can you please attach the latest screenshots? Or are the last ones the latest?

**Comment by garyluu** [comment]
Oh crud, didn't notice this before, but can you move the styles.scss section:

```
// Material Datatable for both tool and workflow list components
...
```
into your newly created scss?

**Comment by coverbeck** [comment]
> Oh crud, didn't notice this before, but can you move the styles.scss section: ...

Please take screenshots after this change. :)

**Comment by ByteMap** [comment]
Here are the requested screenshots @coverbeck 
Screen Shot 2019-08-30 at 2 08 33 PM
Screen Shot 2019-08-30 at 2 08 43 PM
Screen Shot 2019-08-30 at 2 09 01 PM
Screen Shot 2019-08-30 at 2 09 08 PM

**Pull request merged**